src/apify_client/clients/base/actor_job_base_client.py

```diff
@@ -1,13 +1,40 @@
-from typing import Optional
+import math
+from datetime import datetime
+from typing import Dict, Optional
 
+from ..._consts import TERMINAL_ACTOR_JOB_STATUSES, ActorJobStatus
+from ..._utils import _parse_date_fields, _pluck_data
 from .resource_client import ResourceClient
 
-# from ..._utils import _parse_date_fields, _pluck_data
-
 
 class ActorJobBaseClient(ResourceClient):
     """Base sub-client class for actor runs and actor builds."""
 
-    def _wait_for_finish(self, wait_secs: Optional[int] = None) -> None:
-        # TODO implement when implementing run and build endpoints
-        pass
+    def _wait_for_finish(self, wait_secs: Optional[int] = None) -> Dict:
+        started_at = datetime.now()
+        should_repeat = True
+        job: Optional[Dict] = None
+        seconds_elapsed = 0
+
+        while(should_repeat):
+            wait_for_finish = 9999
+            if wait_secs is not None:
+                wait_for_finish = wait_secs - seconds_elapsed
+
+            response = self.http_client.call(
+                url=self.url,
+                method='GET',
+                params=self._params(wait_for_finish=wait_for_finish),
```

Comment:
You have to have `waitForFinish=wait_for_finish` here, the API URL parameter is camel-cased, see https://docs.apify.com/api/v2#/reference/actor-builds/build-object/get-build.

Reply:
I'm changing it, but I don't think this is the case. It worked with `wait_for_finish` as well, I guess the HTTP client must be translating this in payload.

Reply:
Just tested it with locally running API and the snake_case params do not get converted to camelCase params, the `wait_for_finish` param just did not get parsed, so dunno what went on there.